### PR TITLE
initial import of code

### PR DESCRIPTION
**Pull request by theotherian**
## High Level Overview of Change

First attempt at trying to migrate header and error logic from the PayID server to a library

### Context of Change

I'm not super clear on what I'm doing with regard to creating a new TS module for our PayID server, but here we are anyway.

I ran into a few linting issues that were confusing to me, namely regarding exporting things. It seems like it's unhappy with something simply being exported at the `index.ts` level but maybe I just need the `public` modifier to calm it down?

Could use some feedback on correct structure here.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Before: this code was in the PayID server.
After: library 🎉 

## Test Plan

Unit tests have been migrated.

### CODE REVIEW

File: src/errors/getPayIdHttpError.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+/**
+ * A helper function for logging errors and returning a wrapper for the values for the HTTP response.
+ *
+ * @param errorDetails - An HTTP error code and message.
+ * @param logger - The application-level logger.
+ * @param err - The associated error object (optional).
+ *
+ * @returns Details for the HTTP error response to send back to the client.
+ */
+export default function getPayIdHttpError(
```
Comment (dino-rodriguez) [inline src/errors/getPayIdHttpError.ts:31]: Nice, this looks like a good top-level access point.

File: src/headers/headers.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import ParseError, { ParseErrorType } from '../errors/parseError'
```
Comment (dino-rodriguez) [inline src/headers/headers.ts:1]: I'm wondering if headers should live here, or in a separate library. Nonetheless, a good starting point for moving some of this functionality into libraries!

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,2 @@
+* @hbergren
```
Comment (dino-rodriguez) [inline CODEOWNERS:1]:
```suggestion
* @hbergren
* @dino-rodriguez 
```

Comment (dino-rodriguez) [inline CODEOWNERS:1]:
```suggestion
* @hbergren
* @dino-rodriguez 
```

Comment (dino-rodriguez) [reply CODEOWNERS:1]: Adding myself as a CODEOWNER.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    "build": "rm -rf dist && npm run lintNoFix && tsc --project . && npm run test",
+    "lint": "eslint . --ext .ts --fix --max-warnings 2 && prettier --write '**/*.{md,json}'",
+    "lintNoFix": "eslint . --ext .ts --max-warnings 2 && prettier --check '**/*.{md,json}'",
+    "test": "nyc mocha 'test/**/*.test.ts'"
+  },
+  "dependencies": {
+    "@hapi/boom": "^9.1.0",
+    "@xpring-eng/http-status": "^1.0.0",
+    "@xpring-eng/logger": "^1.0.0",
+    "log4js": "^6.3.0"
```
[the remark this answers is not in the document]
Comment (theotherian) [reply package.json:30]: I meant to remove the `xpring-eng/logger` actually, but maybe that's wrong. This comes back to the question I had yesterday around the application vs the library defining the logging level, which maybe needs to be part of a broader discussion next week when Hans is back.

File: src/headers/headers.ts (hunk 2 above)
Comment (theotherian) [reply src/headers/headers.ts:1]: Same; we probably need to discuss general organization of libraries more.

**Comment by theotherian** [comment]
I think this is closer to a final product but I'd like @hbergren to look again. I pulled in more of the code from the PayID Server and kept looking at this library and the server itself to see if what I was doing made sense.

I think the error stuff may end up being marginal at best in the Lambda, but the header stuff will cut down the complexity. 

I still find myself questioning my work though overall for some reason and I can't shake why. It seems like it makes sense to encapsulate this but I keep feeling like I'm somehow making things clunky.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (theotherian) [reply package.json]: I ended up using the xpring logger directly in my code instead. I think it's probably too clunky to try to ask for the logger as an argument.

**Pull request closed without merging**